modules/core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/BulkServiceImpl.java

```diff
@@ -328,4 +329,30 @@ public boolean await(Duration duration) throws InterruptedException {
                  .collect(Collectors.toList());
     }
 
+    @Override
+    public void appendExternalBucket(BulkBucket bucket) {
+        String commandId = bucket.getCommandId();
+
+        BulkCommand command = getCommand(commandId);
```

Comment:
This means a KV store roundtrip for each bucket to get the action name which is a potential perf problem.
A possible solution is to have a very basic cache on getCommand (just keep the last entry), 
or to provide a different method signature appendExternalBucket(String action, BulkBucket bucket), because the scroller should have the knowledge of the action.

Reply:
I chose to use a `PassiveExpiringMap` with a TTL of 1 minute because, in the context of the 10B benchmark, we may have two commands at the same time, one for each repository.